src/schema-contributor.ts

```diff
@@ -43,42 +43,37 @@ class SchemaContributor {
         return true;
     }
 
-    // TODO: Rewrite comments.
-	/**
-	 * Call requestSchema for each provider and find the first one who reports he can provide the schema.
-	 *
-	 * @param {string} resource
-	 * @returns {string} the schema uri
-	 */
-	public requestCustomSchema(resource: string): string {
-        // TODO: This is what gets called on every request(I think), it's looking at the cached schema files. I think here is where we want to periodically
-        //       check what's on the server. The code in schema-association-service.getSchemaAssociation()
-
-        // Check relationship with result of getSchemaAssociationFromYamlValidationNode. Does this load the files specified there? Make sure this code is needed.
-
+    /**
+     * Asks each schema provider whether it has a schema for the given resource,
+     * and returns the URI to the schema if it does.
+     *
+     * @param {string} resource the file to be validated
+     * @returns {string} the schema uri
+     */
+    public requestCustomSchema(resource: string): string {
         for (let customKey of Object.keys(this._customSchemaContributors)) {
             const contributor = this._customSchemaContributors[customKey];
             const uri = contributor.requestSchema(resource);
             if (uri) {
                 return uri;
-            } else {
-                logger.log(`Uri NOT found for resource (${resource})`);
             }
         }
 
+        // TODO: This is currently the only way to fallback to the default schema provider.
+        // The upstream Red Hat server also falls back when receiving a falsy value,
+        // so sync with their changes and change this to return false or something.
         throw `Unable to find custom schema for resource: '${resource}'`;
```

Comment:
FYI since we never register any custom providers (and don't need to), we'll always throw. But it looks like the server forces us to handle this, so ¯\\\_(ツ)\_/¯.